docs/extensions/api.md

```diff
@@ -26,34 +28,51 @@ other issues on the [bug tracker].
 
 ## Preprocessors {: #preprocessors }
 
-Preprocessors munge the source text before it is passed into the Markdown
-parser. This is an excellent place to clean up bad syntax, extract things the
-parser may otherwise choke on and perhaps even store it for later retrieval.
+Preprocessors munge the source text before it is passed to the Markdown
+parser. This is an excellent place to clean up bad encodings or to extract 
+portions for later processing that the parser may otherwise choke on.
 
 Preprocessors should inherit from `markdown.preprocessors.Preprocessor` and
-implement a `run` method with one argument `lines`. The `run` method of
-each Preprocessor will be passed the entire source text as a list of Unicode
-strings. Each string will contain one line of text. The `run` method should
-return either that list, or an altered list of Unicode strings.
-
+implement a `run` method, which takes a single parameter `lines`. This parameter is
+the entire source text stored as a list of Unicode string, one per line.  `run` should
+return its processed list of of Unicode strings, one per line.
 
-A pseudo example:
+A simple example:
 
 ```python
 from markdown.preprocessors import Preprocessor
+import re
 
-class MyPreprocessor(Preprocessor):
+class NoRender(Preprocessor):
+    """ Skip any line with words 'NO RENDER' in it. """
     def run(self, lines):
         new_lines = []
         for line in lines:
-            m = MYREGEX.match(line)
+            m = re.search("NO RENDER", line)
             if m:
-                # do stuff
+                pass   # skipping this line
```

Comment:
To those of us who are well versed in Python, this change is a non-issue. But to newbies, `pass` is confusing. This doesn't improve things at all. I realize that means the code won't run as-is, but that is why we called it "pseudo code" above. Let's keep it that way.

Reply:
I respectfully disagree.

First, this is now code, instead of pseudo-code and does run if it is registered.  The example now does something; removing NO RENDER lines.  Second, `pass` is necessary here.   The `if` syntax requires  a statement  and a comment is not a statement.  Finally, it is reasonable to assume people on their first programming project are not writing Markdown extentions.

Reply:
I am not persuaded. After years of dealing with this sort of thing, the best is what we had. Please change it back.

Reply:
You can also consider using string literals:
```python
if m:
    "do stuff"
```

This will be valid Python code, and maybe less confusing.

Reply:
I'm not sure it has to be valid. It's really just a placeholder for real code. I think we are bikeshedding here. I'm personally indifferent. I'm not sure users are actually getting hung up on this.

Reply:
I could understand fixing it if we were actually getting real world complaints.

Reply:
We still need to remove the `pass`. I'm not approving this otherwise.

Reply:
So I took another look at this and it occurs to me that this could be a working code block without the `pass`. 

Instead of this:

```python
if m:
    pass   # skipping this line
else:
    new_lines.append(line)
```

Do this:

```python
if not m:
    new_lines.append(line)
```

In fact, that is better Python code. So we have two options:

1. Revert to the previous pseudo code
2. Update the example to as per above

Reply:
Good point.  I will change it shortly.